master/internal/user/authz_basic_impl.go

```diff
@@ -108,31 +110,55 @@ func (a *UserAuthZBasic) CanSetUsersDisplayName(
 func (a *UserAuthZBasic) CanCreateAccessToken(
 	ctx context.Context, curUser, targetUser model.User,
 ) error {
-	if !curUser.Admin && curUser.ID != targetUser.ID {
-		return fmt.Errorf("only admin privileged users can create token for other users")
+	if !curUser.Admin {
+		return fmt.Errorf("only admin privileged users can create token for own/other users")
 	}
 	return nil
 }
 
-// CanGetAccessTokens returns an error if the user is not an admin.
+// CanGetAccessTokens returns an error if the user does not have permission to view own or
+// another user's token based on own role.
 func (a *UserAuthZBasic) CanGetAccessTokens(
-	ctx context.Context, curUser, targetUser model.User,
```

Comment:
Do we need more than just the target user's ID, if not would it make sense if we instead passed the the model.AccessTokenInfo record? That it's clearer what this function is looking to accomplished based on it's parameters?

Reply:
Would it be worth potentially modifying this to do the query return so it's applying a filter to the query based on the current user's role?

Reply:
> Would it be worth potentially modifying this to do the query return so it's applying a filter to the query based on the current user's role?

Please help clarify a little more

Reply:
Maybe making it more aligned with how the rbac implementation is working, so it'll take the query and apply the appropriate filter based on if the user is an admin or not

Reply:
Wait, I misread the git diff 🤦